sympy/geometry/point.py

```diff
@@ -266,7 +264,6 @@ def distance(self, p):
         sqrt(x**2 + y**2)
 
         """
-        p = Point(p)
         return sqrt(sum([(a - b)**2 for a, b in zip(self.args, p.args)]))
```

Comment:
maybe this?

```python
try:
    args = p.args
except AttributeError:
    try:
        args = [as_int(I) for I in p]
    except TypeError:
        args = Point(p).args
return sqrt(sum([(a - b)**2 for a, b in zip(self.args, args)]))
```

If this doesn't work then @jksuom suggestion about a keyword may be apropos. Alternatively, you could create a `_distance` method which `distance` calls after making `p` a `Point` -- then that could be called directly if you know you are working with Points.

Or we could "prefer one right way" and make geometry stop accepting tuples as an alias for `Point`. Then we would let the error rise in the routine when `args` are called for from the tuple.